src/Interfaces.jl

```diff
@@ -2767,6 +2767,19 @@ function _to_main(rows::PRange)
   mrows = PRange(ngids,partition)
 end
 
+function Base.show(io::IO, ::MIME"text/plain", A::Union{PSparseMatrix{T,S},PVector{T,S}}) where {T, M, S <: AbstractPData{M}}
+    _prefixstr(A::PSparseMatrix{T}) where T = "$(size(A, 1))×$(size(A,2)) PSparseMatrix{$T}"
+    _prefixstr(A::PVector{T}) where T = "$(length(A))-element PVector{$T}"
+    prefix = _prefixstr(A)
+    np = num_parts(A.values)
+    s = np > 1 ? "s" : ""
+    bt = typeof(get_backend(A.values))
+    iob = IOBuffer()
```

Comment:
I guess that in MPI mode each rank will show the message. We only want the main rank to display info to avoid overcrowding of the output.
Put the lines that do the actual printing in a block like this one:

```julia
map_main(A.values) do values
   print(iob, prefix, " with $np part$s $bt of $M")
   write(io, seekstart(iob))
   nothing
end
 
```

Reply:
> I guess that in MPI mode each rank will show the message.

Yea, but that is how regular objects behave too, so wasn't sure what was the best way to handle it. With the current implementation users can decide for themselves (e.g. put printing behind a `if rank == 0` check or similar), but with the `map_main` suggestion it wouldn't be possible to print from non-rank 0 processes?

Reply:
On the other hand, other objects don't have a way to know if they are rank 0, so perhaps it should be taken advantage of here.

Reply:
Hi @fredrikekre I believe that the output of `display` should be consistent between different backends whenever possible (in this case it is possible).

I agree that most of the use of `display` will be in the REPL with sequential backend, but nothing prevents one to type `display(a)` in  a MPI non interactive computation. That's why I think having consistent output is important.